Common/Data/Consolidators/RangeConsolidator.cs

```diff
@@ -115,39 +123,34 @@ public RangeConsolidator(decimal range)
         }
 
         /// <summary>
-        /// Update the current RangeBar being created with the given data.
+        /// Updates the current RangeBar being created with the given data.
+        /// Additionally, if it's the case, it consolidates the current RangeBar
         /// </summary>
         /// <param name="time">Time of the given data</param>
         /// <param name="currentValue">Value of the given data</param>
         /// <param name="volume">Volume of the given data</param>
         protected override void UpdateBar(DateTime time, decimal currentValue, decimal volume)
         {
-            if (currentValue > CurrentRangeBar.Low + 2 * RangeSize)
+            bool isRising;
+            if (currentValue > CurrentRangeBar.High)
             {
-                Rising(time, currentValue, volume);
+                isRising = true;
             }
-            else if (currentValue < CurrentRangeBar.High - (2 * RangeSize))
+            else if (currentValue < CurrentRangeBar.Low)
             {
-                Falling(time, currentValue, volume);
+                isRising = false;
             }
             else
             {
-                CurrentRangeBar.Update(time, currentValue, volume);
+                return;
             }
-        }
 
-        /// <summary>
-        /// Checks if the current RangeBar being created has closed. If that is the case, it consolidates
-        /// the bar, resets the current RangeBar and stores the needed information from the last bar to
-        /// create the next bar
-        /// <see cref="UpdateBar(DateTime, decimal, decimal)"/>
-        /// </summary>
-        protected override void CheckIfBarIsClosed()
-        {
-            if (CurrentRangeBar != null && CurrentRangeBar.IsClosed)
+            CurrentRangeBar.Update(time, currentValue, volume);
+            while (CurrentRangeBar.IsClosed)
             {
                 OnDataConsolidated(CurrentRangeBar);
-                CurrentRangeBar = null;
+                CurrentRangeBar = new RangeBar(CurrentRangeBar.Symbol, CurrentRangeBar.EndTime, RangeSize, isRising ? CurrentRangeBar.High + _minimumPriceVariation : CurrentRangeBar.Low - _minimumPriceVariation, 0);
+                CurrentRangeBar.Update(time, currentValue, Math.Abs(CurrentRangeBar.Low - currentValue) > RangeSize ? 0 : volume); // Intermediate/phantom RangeBar's have zero volume
```

Comment:
For the phantom bars volume, should this check depend on `isRising`? Something like: if true, `currentValue - CurrentBar.Low` else `CurrentBar.High - currentValue`.

Reply:
Interestingly enough, it doesn't need, because just the bar is just created a line above, so the all its values `Open, High, Low` and `Close` are the same